test/src/test_utils.h

```diff
@@ -5,33 +5,67 @@
  */
 #ifndef TIMESCALEDB_TEST_UTILS_H
 #define TIMESCALEDB_TEST_UTILS_H
-#include <postgres.h>
 
+#include <postgres.h>
 #include <access/xact.h>
 
-#define AssertInt64Eq(a, b)                                                                        \
+#include "export.h"
+
+static inline const char *
+strip_path(const char *filename)
+{
+	int i = 0, slash = 0;
+
+	while (filename[i] != '\0')
+	{
+		if (filename[i] == '/' || filename[i] == '\\')
+			slash = i;
+		i++;
+	}
+
+	return &filename[slash + 1];
+}
+
+#define TestFailure(fmt, ...)                                                                      \
+	do                                                                                             \
+	{                                                                                              \
+		elog(ERROR,                                                                                \
+			 "TestFailure @ %s %s:%d | " fmt "",                                                   \
+			 strip_path(__FILE__),                                                                 \
+			 __func__,                                                                             \
+			 __LINE__,                                                                             \
+			 ##__VA_ARGS__);                                                                       \
```

Comment:
I don't think you need to put the function, file, and line in the error message. They are already included in the logging details if you turn on `VERBOSE`.

Reply:
This is useful in case of a test failure in, e.g., a CI build. We don't want to have to ensure we run with VERBOSE and then save the postgres logs just to figure out where the failure happened.